### DISCUSSION TITLE
Use GH Actions for semantic-release

### DISCUSSION THREAD

**Discussion by okonet** [sapegin/mrm#330] [Ideas]
Some of my Travis CI integrations were deprecated and won't run anymore (not sure who's fault, looks like GH) but I'm wondering if semantic-release should use a GH action instead to avoid breakages (I hope as long it's on GH infra it will run).

**Comment by sapegin** [comment]
I guess so but I don't have any experience with Actions, though I wanted to try. Want to teach me? ;-)

**Comment by okonet** [comment]
I just did it for another repo and it's rather simple. See https://github.com/okonet/react-scroll-sync/pull/28

The only thing that can't be automated ATM is the requirement to add `NPM_TOKEN` in repo's settings on GitHub. Maybe there is a way to automate that too, but I didn't figure it out.

**Comment by sapegin** [comment]
We don't automate it for Travis either ;-/

**Comment by okonet** [comment]
Then it's rather straightforward. I can draft a PR when I have some time.

The only difference between npm and yarn is different ways of installing dependencies: 

- yarn: https://github.com/vieira/homebridge-yeelight-wifi/blob/4215c1a77b1c28a1ab3d53842ac8d82a8496b8df/.github/workflows/release.yml#L16-L19
- npm: https://github.com/okonet/react-scroll-sync/blob/e31b44fea76567d7ed9318c41b573358b14b979c/.github/workflows/release.yml#L17-L18

**Comment by sapegin** [comment]
I think we can detect if the project users Yarn by checking the lock file and support both?
